src/client.cpp

```diff
@@ -142,33 +141,26 @@ ADDON_STATUS ADDON_Create(void* hdl, void* _unused(props))
   ADDON_ReadSettings();
   
   /* Create a settings object that can be used without locks */
-  tvheadend::Settings &settings = tvheadend::Settings::GetInstance();
-  settings.strHostname = g_strHostname;
-  settings.iPortHTSP = g_iPortHTSP;
-  settings.iPortHTTP = g_iPortHTTP;
-  settings.strUsername = g_strUsername;
-  settings.strPassword = g_strPassword;
-  settings.bTraceDebug = g_bTraceDebug;
-  settings.bAsyncEpg = g_bAsyncEpg;
+  Settings &settings = Settings::GetInstance();
+
+  settings.SetHostname(g_strHostname);
+  settings.SetPortHTSP(g_iPortHTSP);
+  settings.SetPortHTTP(g_iPortHTTP);
+  settings.SetUsername(g_strUsername);
+  settings.SetPassword(g_strPassword);
+  settings.SetTraceDebug(g_bTraceDebug);
+  settings.SetAsyncEpg(g_bAsyncEpg);
 
   /* Timeouts are defined in seconds but we expect them to be in milliseconds. */
-  settings.iConnectTimeout = (g_iConnectTimeout * 1000);
-  settings.iResponseTimeout = (g_iResponseTimeout * 1000);
+  settings.SetConnectTimeout(g_iConnectTimeout * 1000);
+  settings.SetResponseTimeout(g_iResponseTimeout * 1000);
 
-  if (g_bPreTunerEnabled)
-  {
-    settings.iTotalTuners = g_iTotalTuners;
-    settings.iPreTuneCloseDelay = g_iPreTunerCloseDelay;
-  }
-  else
-  {
-    /* When we don't want to use predictive tuning */
-    settings.iTotalTuners = 1;
-    settings.iPreTuneCloseDelay = 0;
-  }
+  settings.SetPreTunerEnabled(g_bPreTunerEnabled);
+  settings.SetTotalTuners(g_bPreTunerEnabled ? g_iTotalTuners : 1);
```

Comment:
Similar thing here, IMO `SetTotalTuners` could handle this check internally. It's one of the benefits of setters compared to plain assignment.